bread/layout/components/forms/queryfield.py

```diff
@@ -0,0 +1,201 @@
+        defaultvalue="",
+        placeholder=None,
+        widgetattributes=None,
+        disabled=False,
+        **kwargs,
+    ):
+        kwargs["_class"] = kwargs.get("_class", "") + f" bx--search bx--search--{size}"
+        kwargs["data_search"] = True
+        kwargs["role"] = "search"
+        width = kwargs.get("width", None)
+        if width:
+            kwargs["style"] = kwargs.get("style", "") + f"width:{width};"
+
+        widgetattributes = {
+            "autocomplete": "off",
+            "autofocus": True,
+            "_class": "bx--search-input",
+            "name": "q",
+            "onfocus": "this.setSelectionRange(this.value.length, this.value.length);",
+            **(widgetattributes or {}),
+        }
+
+        normal_inputid = "normal_search__" + hg.html_id(self)
+        normal_input = hg.INPUT(
+            id=normal_inputid,
+            type="text",
+            value=defaultvalue,
+            placeholder=placeholder or _("Search"),
+            **widgetattributes,
+        )
+        normal_icon = Icon(
+            "search",
+            size=16,
+            _class="bx--search-magnifier",
+            id="searchinputicon_normal",
+            aria_hidden="true",
+        )
+        advanced_inputid = "advanced_search__" + hg.html_id(self)
+        advanced_input = hg.If(
+            model,
+            hg.TEXTAREA(
+                defaultvalue,
+                rows=1,
+                id=advanced_inputid,
+                style="padding-top: 1rem;",
+                placeholder=_("Advanced Search (Press Backspace again to exit)"),
+                **widgetattributes,
+            ),
+        )
+        advanced_icon = hg.If(
+            model,
+            Icon(
+                "script",
+                size=16,
+                _class="bx--search-magnifier",
+                id="searchinputicon_advanced",
+                aria_hidden="true",
+            ),
+        )
+        advanced_mode = hg.INPUT(
+            type="hidden",
+            id="searchinput_advancedmode",
+            name="advancedmode",
+            value="off",
+        )
+
+        super().__init__(
+            hg.DIV(
+                hg.DIV(
+                    normal_input,
+                    advanced_input,
+                    advanced_mode,
+                    id="searchinputcontainer",
+                    style="width: 100%;",
+                ),
+                hg.DIV(
+                    normal_icon,
+                    advanced_icon,
+                    id="searchinputicon",
+                ),
+                _close_button(None, {"value": defaultvalue}),
+                **kwargs,
+            ),
+            hg.If(
+                model,
+                hg.SCRIPT(
+                    hg.format(
+                        """
+                document.addEventListener("DOMContentLoaded", () => {{
+                    const inputContainer = document.getElementById('searchinputcontainer');
```

Comment:
This is just for your information:

We can also use ```$(#searchinputcontainer)``` instead of ```document.getElementById('searchinputcontainer')```. We are using bliss.js, not jQuery, so performance is basically the same as using ```document.getElementById``` directly

Reply:
OK. I didn't know we have that too because when I first saw `$`, I thought it would be jQuery so I avoid it.

Reply:
Right, I would also recommend to avoid jQuery.
What we use is https://blissfuljs.com/ which I find much better as it tries to be compatible with vanilla javascript.